scripts/cmake/common.cmake

```diff
@@ -74,45 +74,47 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     add_compiler_flags(-Werror)
     add_compiler_flags(-fstrict-aliasing)
 
-    # The following options are not valid when clang-cl is used.
-    if(NOT MSVC)
-        add_compiler_flags(-pedantic)
-        add_compiler_flags(-pedantic-errors)
-        add_compiler_flags(-fvisibility=hidden)
-    endif()
+    # # The following options are not valid when clang-cl is used.
+    # if(NOT MSVC)
+    #     add_compiler_flags(-pedantic)
+    #     add_compiler_flags(-pedantic-errors)
+    #     add_compiler_flags(-fvisibility=hidden)
+    # endif()
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     #add_compiler_flags(-Wno-unknown-pragmas)
     add_compiler_flags(-Wall)
     add_compiler_flags(-Wextra)
-    add_compiler_flags(-fdiagnostics-show-option)
-    add_compiler_flags(-Wconversion)
-    add_compiler_flags(-Wold-style-cast)
-    add_compiler_flags(-Wfloat-equal)
-    add_compiler_flags(-Wlogical-op)
-    add_compiler_flags(-Wundef)
-    add_compiler_flags(-Wredundant-decls)
-    add_compiler_flags(-Wshadow)
-    add_compiler_flags(-Wstrict-overflow=5)
-    add_compiler_flags(-Wwrite-strings)
-    add_compiler_flags(-Wpointer-arith)
-    add_compiler_flags(-Wcast-qual)
-    add_compiler_flags(-Wformat=2)
-    add_compiler_flags(-Wswitch-default)
-    add_compiler_flags(-Wmissing-include-dirs)
-    add_compiler_flags(-Wcast-align)
-    add_compiler_flags(-Wswitch-enum)
-    add_compiler_flags(-Wnon-virtual-dtor)
-    add_compiler_flags(-Wctor-dtor-privacy)
-    add_compiler_flags(-Wsign-conversion)
-    add_compiler_flags(-Wdisabled-optimization)
-    add_compiler_flags(-Weffc++)
-    add_compiler_flags(-Winvalid-pch)
-    add_compiler_flags(-Wmissing-declarations)
-    add_compiler_flags(-Woverloaded-virtual)
-    add_compiler_flags(-Wunused-but-set-variable)
-    add_compiler_flags(-Wunused-result)
+    add_compiler_flags(-Wpedantic)
+
+    # add_compiler_flags(-fdiagnostics-show-option)
```

Comment:
All these flags aren't covered by ```-Wall``` and ```-Wextra``` - GCC doesn't have anything like the ```-Weverything``` which clang has...